enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,138 @@
+
+### Goals
+
+Provide a command line that upgrades an operator version using OLM.
+
+### Non-Goals
+
+## Proposal
+
+The Operator SDK currently supports running an operator with OLM using the
+`run` command with the `--olm` flag. 
+This enhancement will add a new subcommand, `bundle-upgrade`,
+to the `run` command. This new command, `run bundle-upgrade`, will perform the upgrade of an operator from a previous version to a new specified version.
+
+Currently, the run command has different subcommands: 
+
+ * `run --local` - exists today; converting to a subcommand will be covered by [OSDK-1126](https://issues.redhat.com/browse/OSDK-1126)
+ * `run packagemanifest` - covered by [OSDK-1126](https://issues.redhat.com/browse/OSDK-1126)
+ * `run bundle` - currently being implemented, design document is covered by [run bundle enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md)
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle-upgrade (alias: bu) <bundle-image> --from-index=<> --starting-csv=<> [--channel=<>] [--operator-namespace=<>] [--install-mode=<>]
+```
+
+* Positional arguments:
+    * `bundle-image`
+* Required flags:
+    * `--from-index` 
+    * `--starting-csv`
+* Optional flags:
+    * `--channel` 
+    * `--operator-namespace`
+    * `--install-mode`
+
+
+* `<bundle-image>` is a positional argument that specifies the bundle image that we want to upgrade to
+* `[--from-index=<>]` is a required flag, which  contains the operator we’re upgrading from
+* `--starting-csv=<>` is a required flag
+
+* `[--channel=]` is an optional flag that specifies the channel to upgrade from within a subscription (default channel used if subscription does not contain a channel)
+  the operator. It will default to the `KUBECONFIG` context.
+* `[--operator-namespace=]` is an optional flag that specifies the namespace to install the operator. It will default to the `KUBECONFIG` context. (see run bundle)
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD). (see run bundle)
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamespace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precedence order is:
+    * `OperatorGroup`, if it already exists
+    * `AllNamespaces`, if the CSV supports it
+    * `OwnNamespace`, if the CSV supports it
+    * If the CSV does not support either of these, `--install-mode` flag
+      will be **required**
+
+#### Assumptions
+* OLM is already installed
+* No previous version of the operator is already running
+* An index image exists that has the from version of the operator bundle
+* New version of the operator exists in the cluster 
```

Comment:
What does this mean? Is it just that the `<bundle-image>` and any images it references are pullable by the cluster? Just want to make sure we're precise, because I could see this being interpreted to mean that the new version of the operator is already installed and running, which would be kinda weird :)

Reply:
Absolutely. 
I meant that the provided images should be accessible and pullable in the cluster by kubelet, I rephrased it